Added database backend

rmdluo opened the pull request:
## Description

- added bash script for setting up empty postgres database
- added API to interact with the database
  - files in backend/src/models define the expected table structure
  - files in backend/src/controllers define some (but not necessarily all) CRUD functions for the database
  - backend/src/client.ts does the actual connection to the database
  - backend/index.ts supplies the API endpoints
- **still needs tests** 
- updated frontend API url to localhost db api endpoint, and updated post request to specify JSON header

## Related Issue

Resolves #11 

## Motivation and Context

- need database

## How Has This Been Tested?

- connected it to the (pending PR) frontend and tested saving a recipe, which worked
- needs more rigorous automated tests

## Screenshots (if appropriate):

None

mgmgperson reviewed (changes requested):
I tried running the script, but it seems like there's the error `psql: error: /init_db.sql: No such file or directory`? The shell script probably needs `db_setup_scripts/init_db.sql` or something similar instead of  just `init_db.sql` to get the directory - this is probably a Mac vs. Linux shell issue.

Everything else seems to work, so good job on that!

rmdluo commented:
> I tried running the script, but it seems like there's the error `psql: error: /init_db.sql: No such file or directory`? The shell script probably needs `db_setup_scripts/init_db.sql` or something similar instead of just `init_db.sql` to get the directory - this is probably a Mac vs. Linux shell issue.
> 
> Everything else seems to work, so good job on that!

fixed, just forgot to update the script folder when I moved the files around

mgmgperson reviewed (approved):
lgtm

wandrew004 reviewed (approved):
Looks good! I was able to start the server and test it out. Thanks for working on this!
